core: Fix error when calling $.fn.rules() on a non-form element.

jsclark opened the pull request:
Closes #1770

staabm commented on src/core.js:
```diff
@@ -117,15 +117,14 @@ $.extend( $.fn, {
 
+		// If nothing is selected, return empty object; can't chain anyway
+		if ( element == null || element.form == null ) {
+			return {};
```
Why do you opt for a empty object?

jsclark replied:
I went with an empty object because this method normally returns an object.  I called this out in my bug report (#1770), and I don't feel strongly about `{}` over `undefined`, but I think keeping the return type consistent is reasonable.

staabm replied:
I would prefer the intial return value

staabm commented:
Please add a unit test

jsclark commented:
@staabm I modified one test and added another, is that not sufficient?  I'm happy to add more tests, but I need clarification on what else you're looking for.

staabm commented:
Tests are looking good. Did you verify they fail without your patch?

It doesnt matter how many commit you add. I will take care of it on merge.

jsclark commented:
Yes, I confirmed that they fail without my changes to core.js.  I'll need to modify the when I change the return value from `{}` back to `undefined` though.  I'll make that change tonight or over the weekend.

staabm commented:
Adjusted the last few bits and rebased, merged it. Thanks @jsclark
